qiskit/visualization/pulse/matplotlib.py

```diff
@@ -530,16 +534,21 @@ def _draw_channels(self, ax, output_channels, interp_method, t0, tf, dt, v_max,
 
             else:
                 continue
+
             # plot label
             ax.text(x=0, y=y0, s=channel.name,
                     fontsize=self.style.axis_font_size,
                     ha='right', va='center')
 
-            y0 -= 1
+            # change the y0 offset for removing spacing when a channel has negative values
+            if test1 and self.style.remove_spacing:
+                y0 -= 0.5
```

Comment:
I think this spacing should be flexible. When we draw, for example, a DRAG pulse with very small `beta` value, imaginary part of the waveform has non-zero negative values but actual amplitude is very small. This will generate huge redundant space.

It might be better to replace this line by maximum negative value of the waveform, instead of nonzero-checking. i.e. `0.5 + abs(min(np.nanmin(re), np.nanmin(im)))`.

Reply:
@nkanazawa1989 thanks for your reply, it should be better now

Reply:
thanks for your great work, the PR is very close to merge!  when `test1 and self.style.remove_spacing = True`, generally `amp_min` is zero (because the waveform has no negative value), and this will add y offset of `-1` for waveform with any tiny negative values. what I meant is you can remove `test1`, and just add `min_amp` to the minimum offset of `0.5`.

It could be better to replace `abs(min(np.nanmin(re), np.nanmin(im)))` by `abs(min(0, np.nanmin(re), np.nanmin(im)))` for some corner cases, i.e. dc bias - waveform comprises only positive numbers, but `min_amp` is not zero.

Reply:
@nkanazawa1989 thanks again for your reply,  i replaced the  by `y0=0.5 + abs(min(0, np.nanmin(re), np.nanmin(im)))` as you suggested and i got these two drawing:
![test_schedule](https://user-images.githubusercontent.com/49644465/66650214-5464fa80-ec30-11e9-912a-b9843845eeea.png)
![image2](https://user-images.githubusercontent.com/49644465/66650260-7494b980-ec30-11e9-81b4-b56385830270.png)

A free space is created by the minimum amplitude `abs(min(0, np.nanmin(re), np.nanmin(im)))`
Is it good?
Otherwise when i use the `test1`, this gap is reduced:
![test_schedule](https://user-images.githubusercontent.com/49644465/66651071-6fd10500-ec32-11e9-9a18-2b52c2d7192e.png)
![image2](https://user-images.githubusercontent.com/49644465/66651078-73648c00-ec32-11e9-9260-6789d8e679c1.png)

What do you think?

Reply:
sorry, you're right.  `amp_min` should be normalized by `v_max`.
```python
            amp_min = v_max * abs(min(0, np.nanmin(re), np.nanmin(im)))
```

This could be worth to adding unittest. The pulse in schedule contains very tiny negative values.
```python
gp0 = pulse_lib.drag(duration=20, amp=1.0, sigma=1.0, beta=0.0001)
acquire = Acquire(10)

sched = Schedule()
sched = sched.insert(30, gp0(device.drives[0]))
sched = sched.insert(50, gp0(device.drives[0]))
sched = sched.insert(70, gp0(device.drives[0]))
sched = sched.insert(90, gp0(device.drives[0]))
sched = sched.insert(90, acquire(device.acquires[1],
                                 device.memoryslots[1],
                                 device.registers[1]))
```

with `test1`, it generates redundant space under d0.
![with_test1](https://user-images.githubusercontent.com/39517270/66693656-b7f93180-ecdd-11e9-9ae2-2bcc321d81ce.png)

with correct `amp_min`, the space will be removed.
![no_test1](https://user-images.githubusercontent.com/39517270/66693666-d9f2b400-ecdd-11e9-9a05-818c5fd2b40f.png)

Reply:
@nkanazawa1989, thank you that's look very better now, could you check again?